lib/ops/opsservice/shrink.go

```diff
@@ -307,8 +307,16 @@ func (s *site) shrinkOperationStart(ctx *operationContext) (err error) {
 		if !force {
 			return trace.Wrap(err, "failed to remove the node from the cluster")
 		}
-		ctx.Warningf("Failed to remove node %q from the cluster, force continue: %v.",
-			serverName, trace.DebugReport(err))
+		ctx.WithError(err).Warnf("Failed to remove node %q from the cluster, force continue.",
+			serverName)
+	}
+
+	if err = s.removeObjectPeer(server.ObjectPeerID()); err != nil && !trace.IsNotFound(err) {
```

Comment:
Isn't gravity-site still running at this point (when removing a master)? So there's a chance it will just resync the object and put itself as a peer again?